### ISSUE TITLE
I'm unable to reset password on mobile devices

### ISSUE THREAD

**Issue by Pruspat** [ory/kratos#2628]
### Preflight checklist

- [X] I could not find a solution in the existing issues, docs, nor discussions.
- [X] I agree to follow this project's [Code of Conduct](https://github.com/ory/kratos/blob/master/CODE_OF_CONDUCT.md).
- [X] I have read and am following this repository's [Contribution Guidelines](https://github.com/ory/kratos/blob/master/CONTRIBUTING.md).
- [ ] This issue affects my [Ory Cloud](https://www.ory.sh/) project.
- [X] I have joined the [Ory Community Slack](https://slack.ory.sh).
- [X] I am signed up to the [Ory Security Patch Newsletter](https://ory.us10.list-manage.com/subscribe?u=ffb1a878e4ec6c0ed312a3480&id=f605a41b53).

### Describe the bug

I'm not able to finish account recovery on mobile devices?

1. I'm calling GET recovery/api
2. then in the POST request I pass the email address and method "link"
3. I receive an email message with a URL like: "http://127.0.0.1:4433/self-service/recovery?flow=91b83275-f306-463f-92b3-9052cae8ec23&token=Z5ehZKrk7sWaFBRxi8aI5xDlgYViXaWO"

In a web application, if you will GET the URL below, you will receive cookies which makes it possible to init settings and then change the password, but I'm unable to receive a session token for mobile devices to init setting and then change my password

### Reproducing the bug

Run steps that are above

### Version

oryd/kratos v0.8.2-alpha.1 and oryd/kratos:v0.10.0

### On which operating system are you observing this issue?

Linux

### In which environment are you deploying?

Docker Compose

**Comment by aeneasr** [comment]
Hi, thank you for the report! That is indeed the case - there are some open PRs for this but in general we have plans to switch to a code which people will enter in their app, making this process mobile friendly!

For now, your users will have to change the password in the browser :/

**Comment by kmherrmann** [comment]
@jonas-jonas JFYI as you're drilling into API flows

**Comment by sobanieca-redocly** [comment]
Hello, I just want to point out that I've run into issue where I'm missing this functionality. I'm creating recovery flow via Native Apps endpoint, but I got stuck when I've validated recovery code, as in result I'm just getting 422 response with cookies for browser redirect to configure new password. It would be awesome if for native apps flow we would receive session token when proper recovery code was sent and using it, we could call settings flow endpoint to set a new password.
